### ISSUE TITLE
Updating torch lightning

### ISSUE THREAD

**Issue by RaulPPelaez** [torchmd/torchmd-net#168]
In a quest to use the latest versions of the different dependencies (mainly for torch2) I have encountered an issue with torch-lightning.
The API has changed in some slight ways, in particular this reset line here has been deprecated:
https://github.com/torchmd/torchmd-net/blob/e02c076aefea88ed39a5b161ba48113b31ba331e/torchmdnet/module.py#L160-L163
There is a succint message refering to this in the docs:
```markdown
   * - used ``trainer.reset_*_dataloader()`` methods
     - use  ``Loop.setup_data()`` for the top-level loops
     - `PR16726`_
```
Pointing to this PR: https://github.com/Lightning-AI/lightning/pull/16726

I do not know, however, what this line is doing exactly and I do not understand the comment.
Do you have some pointers?
cc @PhilippThoelke

**Comment by PhilippThoelke** [comment]
To evaluate the test set every couple of epochs during training we add it as a val_dataloader. In the test epochs we then have two dataloaders (validation + test) instead of only the validation dataloader. The deprecated function resets the Trainer's internal state of which data to use for what, allowing us to evaluate on the test set only every couple of epochs but evaluate on the validation set every epoch. You'd have to make sure that the suggested replacement method achieves the same thing without breaking or slowing down anything else.
